test cases/unit/59 introspect buildoptions/meson.build

```diff
@@ -9,7 +9,7 @@ foo = false
 
 executable(target_name, target_src, build_by_default : foo)
 
-r = run_command(find_program('c_compiler.py'))
+r = run_command(find_program('c_compiler.py'), check: false)
 if r.returncode() != 0
   error('FAILED')
```

Comment:
This one should should probably be check: true, lol

Reply:
Make sure we still have at least one place where we test with `check: false` and a command that actually fails.

Reply:
`33 run program` does that several times. ` assert(ret.returncode() == 0, 'failed to run python3: ' + ret.stderr())`

If you're looking for one that is expected to error out and raises a testsuite failure if `ret.returncode() == 0`, this doesn't do that either.

Reply:
If you run `rg -F "check: false" test\ cases/`, you'll find this pattern at a couple of places:
```
r = run_command(..., check: false)
if r.returncode() != 0
    error('some error')
endif
```
(sometimes with an if clause, sometimes with an assert). This can be removed if we set `check` to `true`, but this would change the error message.
In some places (like `error('FAILED')`), this error message isn't better than the default, but in some other places it is.
I think you should decide which places should keep their custom error messages and which places can use the default error message.